src/Persistence/Search/Query/ItemTypeQueryBuilder.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Persistence\Search\Query;
+
+use Elastica\Query;
+use Elastica\Query\AbstractQuery;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+
+class ItemTypeQueryBuilder {
+
+	public function __construct(
+		private readonly PropertyId $itemTypeProperty
+	) {
+	}
+
+	public function buildQuery( array $itemTypes ): AbstractQuery {
+		return new Query\Terms(
```

Comment:
Is this partial import on purpose?

Reply:
Should the return type be `Terms` from `Elastica\Query\Terms` instead of `AbstractQuery`?

Reply:
> Is this partial import on purpose?

It was kind of, since I this pattern being used in a few places for Elastica query objects. But I suppose this is arbitrary and WikibaseCirrusSearch imports them directly, so I'll do that instead.

Reply:
> Should the return type be Terms from Elastica\Query\Terms instead of AbstractQuery?

No. All the query objects extend AbstractQuery. The calling code does not need to know what kind of query is constructed. It just needs to know the most narrow interface.